library/core/src/result.rs

```diff
@@ -1626,6 +1626,25 @@ impl<A, E, V: FromIterator<A>> FromIterator<Result<A, E>> for Result<V, E> {
     }
 }
 
+//FIXME: Figure out how to mark this as unstable without x.py refusing to run any tests
+#[stable(feature = "tuple_from_result_iter", since = "1.53.0")]
+impl<T, E, U, V> FromIterator<Result<T, E>> for (U, V)
+where
+    U: Default + Extend<T>,
+    V: Default + Extend<E>,
+{
+    fn from_iter<I: IntoIterator<Item = Result<T, E>>>(iter: I) -> (U, V) {
+        let (mut oks, mut errs) = (U::default(), V::default());
+        for result in iter {
```

Comment:
Couldn't this use `for_each`?

Reply:
Just wondering, why did this got a negative feedback? Internal iteration should always be either as fast as or faster than external iteration, so you should always try to use it if you can, hence why I would have used `for_each` instead of a normal `for` loop here.

Reply:
> Internal iteration should always be either as fast as or faster than external iteration, so you should always try to use it if you can

I disagree with this. for_each introduces a lot more syntactic noise and does the same thing as a normal for loop. If this happens to be super slow for some reason it seems fine to use for_each, but I don't think it should be the default.

Reply:
Also, you didn't mention at first that this was for performance reasons, so I thought it was just personal preference.

Reply:
> If this happens to be super slow for some reason it seems fine to use for_each, 

It's already pretty slow for iterators like `RangeInclusive` and `VecDeque`, and probably many iterator adapters that need to do additional checks in each iteration.

> but I don't think it should be the default.

But isn't it already the default in pretty much all the stdlib? `Iterator`'s utility methods and any implementation of `FromIterator`, `Extend`, `Sum` and `Product` all use internal iteration methods. The only occurences of plain `for` loops I can find are where performance doesn't matter (for example tests) or when the iterator doesn't gain anything with internal iteration (for example exclusive ranges, slices and vecs).

> Also, you didn't mention at first that this was for performance reasons, so I thought it was just personal preference.

Yeah my bad for that one, I was assuming that the difference between `for` and `for_each` is only performance.